pkg/clients/frontegg_client.go

```diff
@@ -0,0 +1,238 @@
+}
+
+// GetToken function to authenticate with the Frontegg API and retrieve a token
+func getToken(ctx context.Context, password string, endpoint string) (string, string, time.Time, error) {
+	clientId, secretKey, err := parseAppPassword(password)
+	if err != nil {
+		return "", "", time.Time{}, err
+	}
+
+	adminEndpoint := fmt.Sprintf("%s/identity/resources/auth/v1/api-token", endpoint)
+
+	payload := map[string]string{
+		"clientId": clientId,
+		"secret":   secretKey,
+	}
+	payloadBytes, err := json.Marshal(payload)
+	if err != nil {
+		return "", "", time.Time{}, err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", adminEndpoint, bytes.NewBuffer(payloadBytes))
+	if err != nil {
+		return "", "", time.Time{}, err
+	}
+	req.Header.Add("Content-Type", "application/json")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return "", "", time.Time{}, err
+	}
+	defer resp.Body.Close()
+
+	// Read the response body into the 'body' variable
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", "", time.Time{}, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return "", "", time.Time{}, fmt.Errorf("authentication failed: %s", string(body))
+	}
+
+	var result map[string]interface{}
+	if err := json.Unmarshal(body, &result); err != nil {
+		return "", "", time.Time{}, err
+	}
+
+	tokenString, ok := result["accessToken"].(string)
+	if !ok {
+		return "", "", time.Time{}, errors.New("access token not found in the response")
+	}
+
+	// Parse the token without verifying the signature.
+	token, _, err := new(jwt.Parser).ParseUnverified(tokenString, jwt.MapClaims{})
+	if err != nil {
+		return "", "", time.Time{}, fmt.Errorf("error parsing token: %v", err)
+	}
+
+	claims, ok := token.Claims.(jwt.MapClaims)
+	if !ok {
+		return "", "", time.Time{}, errors.New("invalid token claims")
+	}
+
+	email, ok := claims["email"].(string)
+	if !ok {
+		return "", "", time.Time{}, errors.New("email claim not found in token")
+	}
+
+	var tokenExpiry time.Time
+	if expiresIn, ok := result["expiresIn"].(float64); ok {
+		tokenExpiry = time.Now().Add(time.Duration(expiresIn) * time.Second)
+	} else {
+		// Default expiry time if not provided in the response
+		tokenExpiry = time.Now().Add(1 * time.Hour)
+	}
+
+	return tokenString, email, tokenExpiry, nil
+}
+
+// Get the token from the FronteggClient
+func (c *FronteggClient) GetToken() (string, error) {
+	return c.Token, nil
+}
+
+// Get the email from the FronteggClient
+func (c *FronteggClient) GetEmail() (string, error) {
+	return c.Email, nil
+}
+
+// Get the endpoint from the FronteggClient
+func (c *FronteggClient) GetEndpoint() (string, error) {
+	return c.Endpoint, nil
+}
+
+// Get the token expiry from the FronteggClient
+func (c *FronteggClient) GetTokenExpiry() (time.Time, error) {
+	return c.TokenExpiry, nil
+}
+
+// Get the password from the FronteggClient
+func (c *FronteggClient) GetPassword() (string, error) {
+	return c.Password, nil
+}
```

Comment:
Why do these return an error if it's always nil?

Reply:
Ah yes, seems like I've missed this after one of the reactors! Good catch.